<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Materials | A Gentle Introduction to Spatial Transcriptomics Analysis</title>
  <meta name="description" content="Chapter 3 Materials | A Gentle Introduction to Spatial Transcriptomics Analysis" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Materials | A Gentle Introduction to Spatial Transcriptomics Analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Materials | A Gentle Introduction to Spatial Transcriptomics Analysis" />
  
  
  

<meta name="author" content="Jessica Gillespie" />


<meta name="date" content="2025-04-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="qc-clustering-and-annotating-with-seurat.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome!</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#who-is-this-not-for"><i class="fa fa-check"></i><b>1.1</b> Who is this NOT for</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#so-who-is-this-actually-for-then"><i class="fa fa-check"></i><b>1.2</b> So who is this actually for then?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#want-to-contribute"><i class="fa fa-check"></i><b>1.3</b> Want to contribute?</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#about-this-actual-site"><i class="fa fa-check"></i><b>1.4</b> About this actual site</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#general-workflow"><i class="fa fa-check"></i><b>2.1</b> General Workflow</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="materials.html"><a href="materials.html"><i class="fa fa-check"></i><b>3</b> Materials</a>
<ul>
<li class="chapter" data-level="3.1" data-path="materials.html"><a href="materials.html#computer-infrastructure"><i class="fa fa-check"></i><b>3.1</b> Computer Infrastructure</a></li>
<li class="chapter" data-level="3.2" data-path="materials.html"><a href="materials.html#software"><i class="fa fa-check"></i><b>3.2</b> Software</a></li>
<li class="chapter" data-level="3.3" data-path="materials.html"><a href="materials.html#collect-necessary-files"><i class="fa fa-check"></i><b>3.3</b> Collect necessary files</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="materials.html"><a href="materials.html#space-ranger-output"><i class="fa fa-check"></i><b>3.3.1</b> Space Ranger output</a></li>
<li class="chapter" data-level="3.3.2" data-path="materials.html"><a href="materials.html#reference-data"><i class="fa fa-check"></i><b>3.3.2</b> Reference data</a></li>
<li class="chapter" data-level="3.3.3" data-path="materials.html"><a href="materials.html#example-datasets"><i class="fa fa-check"></i><b>3.3.3</b> Example datasets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html"><i class="fa fa-check"></i><b>4</b> QC, Clustering, and Annotating with Seurat</a>
<ul>
<li class="chapter" data-level="4.1" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#import-and-qc"><i class="fa fa-check"></i><b>4.1</b> Import and QC</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#load-in-the-data"><i class="fa fa-check"></i><b>4.1.1</b> Load in the data</a></li>
<li class="chapter" data-level="4.1.2" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#trim-and-filter"><i class="fa fa-check"></i><b>4.1.2</b> Trim and filter</a></li>
<li class="chapter" data-level="4.1.3" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#normalization"><i class="fa fa-check"></i><b>4.1.3</b> Normalization</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#visualize-gene-expression"><i class="fa fa-check"></i><b>4.2</b> Visualize Gene Expression</a></li>
<li class="chapter" data-level="4.3" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#dimention-reduction-and-cluster"><i class="fa fa-check"></i><b>4.3</b> Dimention Reduction and Cluster</a></li>
<li class="chapter" data-level="4.4" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#visualizing-clusters"><i class="fa fa-check"></i><b>4.4</b> Visualizing Clusters</a></li>
<li class="chapter" data-level="4.5" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#spot-annotating"><i class="fa fa-check"></i><b>4.5</b> Spot Annotating</a></li>
<li class="chapter" data-level="4.6" data-path="qc-clustering-and-annotating-with-seurat.html"><a href="qc-clustering-and-annotating-with-seurat.html#save-the-data"><i class="fa fa-check"></i><b>4.6</b> Save the data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="spatially-variable-genes-svgs-discovery-with-spark-x.html"><a href="spatially-variable-genes-svgs-discovery-with-spark-x.html"><i class="fa fa-check"></i><b>5</b> Spatially Variable Genes (SVGs) Discovery with SPARK-X</a>
<ul>
<li class="chapter" data-level="5.1" data-path="spatially-variable-genes-svgs-discovery-with-spark-x.html"><a href="spatially-variable-genes-svgs-discovery-with-spark-x.html#importing-and-preparing-data"><i class="fa fa-check"></i><b>5.1</b> Importing and Preparing Data</a></li>
<li class="chapter" data-level="5.2" data-path="spatially-variable-genes-svgs-discovery-with-spark-x.html"><a href="spatially-variable-genes-svgs-discovery-with-spark-x.html#run-spark-x"><i class="fa fa-check"></i><b>5.2</b> Run SPARK-X</a></li>
<li class="chapter" data-level="5.3" data-path="spatially-variable-genes-svgs-discovery-with-spark-x.html"><a href="spatially-variable-genes-svgs-discovery-with-spark-x.html#visualizing-svgs"><i class="fa fa-check"></i><b>5.3</b> Visualizing SVGs</a></li>
<li class="chapter" data-level="5.4" data-path="spatially-variable-genes-svgs-discovery-with-spark-x.html"><a href="spatially-variable-genes-svgs-discovery-with-spark-x.html#save-data"><i class="fa fa-check"></i><b>5.4</b> Save Data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="cell-cell-communication-with-cellchat.html"><a href="cell-cell-communication-with-cellchat.html"><i class="fa fa-check"></i><b>6</b> Cell-Cell Communication with CellChat</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cell-cell-communication-with-cellchat.html"><a href="cell-cell-communication-with-cellchat.html#importing-data-and-setting-computer-options"><i class="fa fa-check"></i><b>6.1</b> Importing Data and Setting Computer Options</a></li>
<li class="chapter" data-level="6.2" data-path="cell-cell-communication-with-cellchat.html"><a href="cell-cell-communication-with-cellchat.html#preparing-cellchat-object-and-database"><i class="fa fa-check"></i><b>6.2</b> Preparing CellChat Object and Database</a></li>
<li class="chapter" data-level="6.3" data-path="cell-cell-communication-with-cellchat.html"><a href="cell-cell-communication-with-cellchat.html#run-cellchat"><i class="fa fa-check"></i><b>6.3</b> Run CellChat</a></li>
<li class="chapter" data-level="6.4" data-path="cell-cell-communication-with-cellchat.html"><a href="cell-cell-communication-with-cellchat.html#visualization"><i class="fa fa-check"></i><b>6.4</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="future-directions-and-conclusion.html"><a href="future-directions-and-conclusion.html"><i class="fa fa-check"></i><b>7</b> Future Directions and Conclusion</a>
<ul>
<li class="chapter" data-level="7.1" data-path="future-directions-and-conclusion.html"><a href="future-directions-and-conclusion.html#deconvolution"><i class="fa fa-check"></i><b>7.1</b> Deconvolution</a></li>
<li class="chapter" data-level="7.2" data-path="future-directions-and-conclusion.html"><a href="future-directions-and-conclusion.html#conclusion"><i class="fa fa-check"></i><b>7.2</b> Conclusion</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Gentle Introduction to Spatial Transcriptomics Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="materials" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Materials<a href="materials.html#materials" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="computer-infrastructure" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Computer Infrastructure<a href="materials.html#computer-infrastructure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As R will be the main working environment any operating system capable of running R is sufficient. ST datasets can be quite large (10’s of GB) and saving data at several stages in the process may be desired so 10-20GB of space may be desired. A multi-core processor for faster analysis may be beneficial at some steps, though not required. Finally, a fairly large amount of RAM may be required (&gt;16GB). High Performance Computing services are available allowing users to analyze their data on a cluster, although that procedure is not detailed here.</p>
</div>
<div id="software" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Software<a href="materials.html#software" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>R is the coding environment in which the analysis will take place. R can be run via the command-line or used in the RStudio GUI. Packages can then be installed from within R. Version numbers used in this protocol are provided but these specific versions are not mandatory.</p>
<ul>
<li><a href="https://www.r-project.org/">R (v4.4.2)</a></li>
<li><a href="https://posit.co/downloads/">RStudio (RStudio 2024.12.0+467 “Kousa Dogwood”)</a></li>
<li><a href="https://satijalab.org/seurat/">Seurat (v5.2.0)</a></li>
<li><a href="https://xzhoulab.github.io/SPARK/">SPARK-X</a></li>
<li><a href="https://github.com/jinworks/CellChat">CellChat (v2.1.0)</a></li>
</ul>
</div>
<div id="collect-necessary-files" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Collect necessary files<a href="materials.html#collect-necessary-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="space-ranger-output" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Space Ranger output<a href="materials.html#space-ranger-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After sequencing is completed, 10X Visium data will be provided as either base call files (.BCL) or FASTQ files, either of which can be used for input to 10X Genomics’ Space Ranger software. The output of Space Ranger is the input for our first analysis method so we will start there. Space Ranger provides a variety of files containing data and various metrics on the samples and sequencing run. We are interested in one file and one folder. We need the filtered feature-barcode matrices: hdf5 which holds the barcodes, features, and RNA count data. We will also need the spatial folder which contains data pertaining to the image that accompanies the sample. For a full picture of output file structure, see the <a href="https://www.10xgenomics.com/support/software/space-ranger/latest/analysis/outputs/output-overview">10X Genomics website</a>.</p>
</div>
<div id="reference-data" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Reference data<a href="materials.html#reference-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In order to mark cell types, we need to compare gene expression to that of already determined cells. This is frequently done with a reference dataset. While the software mentioned here has some reference sets built in, there are others available for download. These can be used after the clustering step for cell-type annotation and in deconvolution. Some annotated datasets are available from the <a href="https://azimuth.hubmapconsortium.org/">HuBMAP Consortium</a>.</p>
</div>
<div id="example-datasets" class="section level3 hasAnchor" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Example datasets<a href="materials.html#example-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>10X Genomics has freely available many datasets produced with their technology for users to download and analyze. These are useful for practice and learning. A mouse anterior brain dataset will be used in this chapter and is available <a href="https://www.10xgenomics.com/datasets/mouse-brain-serial-section-2-sagittal-anterior-1-standard">here</a>. NOTE: If you click this link from an institution’s network you will be asked to sign up for an account. However, if you click this link from a home network, you won’t have to provide an email to download a free dataset.</p>
<p><strong>To follow along with this tutorial</strong> <br>
After clicking the link, scroll down on the page and click on “Output and supplemental files”. Download the “Feature / barcode matrix HDF5 (filtered)” and place it in the <code>datasets</code> folder. Also download the “Spatial imagine data”, a compressed folder named “V1_Mouse_Brain_Sagittal_Anterior_Section_2_spatial.tar.gz” into the <code>dtasets</code> folder. Extracting this should give you a folder named <code>spatial</code> with various files inside. Leave this as is for now.
<br>
<br>
<br></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="materials.html#cb5-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.4.3 (2025-02-28)
## Platform: x86_64-pc-linux-gnu
## Running under: Linux Mint 21
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: America/New_York
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] digest_0.6.37     R6_2.6.1          bookdown_0.42     fastmap_1.2.0    
##  [5] xfun_0.51         cachem_1.1.0      knitr_1.49        htmltools_0.5.8.1
##  [9] rmarkdown_2.29    lifecycle_1.0.4   cli_3.6.4         sass_0.4.9       
## [13] jquerylib_0.1.4   compiler_4.4.3    rstudioapi_0.17.1 tools_4.4.3      
## [17] evaluate_1.0.3    bslib_0.9.0       yaml_2.3.10       jsonlite_1.9.0   
## [21] rlang_1.1.5</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="qc-clustering-and-annotating-with-seurat.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-Materials.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
